.github/workflows/ci.yml

```diff
@@ -36,6 +36,14 @@ jobs:
       matrix:
         python-version: [3.8, 3.9, '3.10', '3.11', '3.12-dev']
         os: ['windows-latest', 'ubuntu-22.04', 'macos-11']
+        extra: [null]
+        include:
+          - python-version: '3.12-dev'
+            os: ubuntu-22.04
+            extra: null
+          - python-version: '3.12-dev'
+            os: ubuntu-22.04
+            extra: completion
```

Comment:
There's already a separate `test-alpine` job hiding at the bottom of this file. Why not install `argcomplete` unconditionally for this job and not install it for the Alpine job? Then we'd cover both the *have argcomplete* and *don't have argcomplete* without CI becoming any slower than it already is.

Reply:
I could make that change if you want; but it looks like the `test-alpine` job is significantly different from these others tests, because it uses Docker. In fact, in order to make that change I would need to change the `alpine.dockerfile`, not this `ci.yml` file -- it would be very easy to not notice that the Python dependencies are intentionally not quite the same. It also introduces another variable to this one-off job; if the tests fail on Alpine, is it because of the Alpine OS, because of Docker, or because of `argcomplete`?

Personally, I think that adding another runner in this matrix is a better solution, for clarity and readability. That said, if you feel that changing `test-alpine` is better, I'll make that change. Let me know what you think.

Reply:
You're reading it backwards. I want the normal jobs to install `argcomplete` and the Alpine job not to so you shouldn't have to touch the dockerfile.

I'm aware it's less obvious what's going on than with an explicit separate job but those 3.12-dev jobs will soon cease to be -dev jobs at which point they'll become 2 hours each. That's a lot of CI time to condemn to the void just to cover a few `except` blocks.

Reply:
What are we actually testing for, though? The fact that `argcomplete.autocomplete` is not broken (or that it is not broken in combination with our CLI definitions)? Surely we do not need to replicate the whole CI run for that...

So I agree with Brénainn's suggestion that `argcomplete` is unconditionally installed on the "main" CI (if it ends up broken we'll have to deal with it one way or another, anway).

Reply:
Fair enough! I've pushed a commit to make that change.